semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,477 @@
+        Uses change in porevolume from Eclipse (RPORV in .UNRST) as input to
+        Geertsma model.
+
+        :param vintage_pairs:
+        """
+        return self._geertsma_ts_custom(
+            vintage_pairs, self.subsidence.eval_geertsma_rporv, "TS_RPORV"
+        )
+
+    def geertsma_ts_simple(self, vintage_pairs):
+        """
+        Calculates TS without using velocity. Fast.
+        Velocity is only used to get the surface on the velocity grid.
+
+        :param vintage_pairs:
+        """
+        return self._geertsma_ts_custom(
+            vintage_pairs, self.subsidence.evalGeertsma, "TS_SIMPLE"
+        )
+
+    def _geertsma_ts_custom(self, vintage_pairs, subsidence_func, method_name):
+        """
+        Calculates TS without using velocity. Fast.
+
+        :param vintage_pairs:
+        :param subsidence_func: specify subsidence method to be used
+        :param method_name: string represeting the subsudence func name
+        """
+
+        if len(vintage_pairs) < 1:
+            return 0, []
+
+        vintages = self._vintages_name_date(vintage_pairs)
+        surface = self._surface
+        points_to_calculate = self._get_non_nan_points()
+        surf_displacement = np.zeros((len(vintages.name), len(surface)))
+        ts_surfaces = []
+        for idx_vtg, (name_vtg, date_vtg) in enumerate(zip(vintages.name, vintages.date)):
+            try:
+                logging.info(
+                    "{:%x %X} {}: Calculating vintage {:%Y.%m.%d}".format(
+                        dt.now(), method_name, date_vtg
+                    )
+                )
+            except TypeError:
+                logging.exception("Wrong date vintage format!")
+                raise ValueError
+
+            self.add_survey(name_vtg, date_vtg)
+
+            for point in points_to_calculate:
+                r1 = (surface.x[point], surface.y[point], 0)
+                r2 = (
+                    surface.x[point],
+                    surface.y[point],
+                    surface.z[point] - self._seabed,
+                )
+                # subsidence and displacement have opposite sign
+                # should have minus on dz1 and dz2 here, more efficient
+                # when calculating ts
+                dz1 = subsidence_func(
+                    base_survey=name_vtg,
+                    monitor_survey=None,
+                    pos=r1,
+                    youngs_modulus=self._youngs_modulus,
+                    poisson_ratio=self._poisson_ratio,
+                    seabed=self._seabed,
+                )
+                dz2 = subsidence_func(
+                    base_survey=name_vtg,
+                    monitor_survey=None,
+                    pos=r2,
+                    youngs_modulus=self._youngs_modulus,
+                    poisson_ratio=self._poisson_ratio,
+                    seabed=self._seabed,
+                )
+                surf_displacement[idx_vtg, point] = dz2 - dz1
```

Comment:
Should perhaps be a separate function?

Reply:
Same as above, ie. don't see any benefit of having this as separate function. Besides, before line `263` there is only initialization of variables. Not saying that I cannot be convinced though :)